packages/ingest/test/test-granule.js

```diff
@@ -553,3 +608,240 @@ test('handleDuplicateFile calls S3.moveObject with expected arguments if duplica
+  t.is(listObjectsResponse.Contents[0].Key, key);
+
+  const pgFile = await filePgModel.search(t.context.knex, {
+    granule_cumulus_id: cumulus_id,
+    file_name: key,
+  });
+
+  t.is(pgFile.length, 1);
+  t.like(pgFile[0], {
+    bucket: secondBucket,
+    key,
+  });
+});
+
+test('moveGranuleFile moves a granule file and does not update postgres when writeToPostgres is false', async (t) => {
+  const bucket = t.context.internalBucket;
+  const secondBucket = t.context.destBucket;
+  const testPrefix = cryptoRandomString({ length: 10 });
+  const fileName = cryptoRandomString({ length: 10 });
+  const key = `${testPrefix}/${fileName}`;
+
+  const filePgModel = new FilePgModel();
+
+  const moveFileParam = {
+    source: {
+      Bucket: bucket,
+      Key: key,
+    },
+    target: {
+      Bucket: secondBucket,
+      Key: key,
+    },
+    file: {
+      bucket,
+      key,
+      name: key,
+    },
+  };
+
+  const params = { Bucket: bucket, Key: key, Body: randomString() };
+  await S3.s3PutObject(params);
+
+  const result = await moveGranuleFile(
+    moveFileParam,
+    filePgModel,
+    t.context.knex,
+    undefined,
+    false
+  );
+
+  t.deepEqual({
+    bucket: secondBucket,
+    key,
+    name: key,
+  }, result);
+
+  const listObjectsResponse = await s3().listObjects({
+    Bucket: secondBucket,
+    Prefix: testPrefix,
+  }).promise();
+  t.is(listObjectsResponse.Contents.length, 1);
+  t.is(listObjectsResponse.Contents[0].Key, key);
+
+  const pgFile = await filePgModel.search(t.context.knex, {
+    file_name: key,
+  });
+
+  t.is(pgFile.length, 0);
+});
+
+test('moveGranuleFile throws when writeToPostgres is true but postgresCumulusGranuleId is defined', async (t) => {
+  const bucket = t.context.internalBucket;
+  const secondBucket = t.context.destBucket;
+  const testPrefix = cryptoRandomString({ length: 10 });
+  const fileName = cryptoRandomString({ length: 10 });
+  const key = `${testPrefix}/${fileName}`;
+
+  const filePgModel = new FilePgModel();
+
+  const moveFileParam = {
+    source: {
+      Bucket: bucket,
+      Key: key,
+    },
+    target: {
+      Bucket: secondBucket,
+      Key: key,
+    },
+    file: {
+      bucket,
+      key,
+      name: key,
+    },
+  };
+
+  await t.throwsAsync(moveGranuleFile(
+    moveFileParam,
+    filePgModel,
+    t.context.knex,
+    undefined,
+    true
+  ));
+});
+
+test('moveGranuleFile returns the expected MovedGranuleFile object if a source and target is missing from the moveFileParams', async (t) => {
+  const bucket = t.context.internalBucket;
+  const testPrefix = cryptoRandomString({ length: 10 });
+  const fileName = cryptoRandomString({ length: 10 });
+  const key = `${testPrefix}/${fileName}`;
+
+  const moveFileParam = {
+    file: {
+      bucket,
+      key,
+      name: key,
+    },
+  };
+
+  const actual = await moveGranuleFile(
+    moveFileParam,
+    undefined,
+    undefined,
+    undefined,
+    false
+  );
+
+  t.deepEqual(actual, {
```

Comment:
So in this instance, moveGranuleFile just returns back the input?

Reply:
Correct - that's the original logic and preserves the behavior for updating dynamoDb.